examples/howto/Model_Builder.myst.md

```diff
@@ -0,0 +1,251 @@
+
+```{code-cell} ipython3
+data, model_config, sampler_config = LinearModel.create_sample_input()
+model = LinearModel(model_config, sampler_config, data)
+```
+
+After making the object of class `LinearModel` we can fit the model using the `.fit()` method.
+
++++
+
+`fit` method takes one argument `data` on which we need to fit the model and assigns `idata.attrs` with `id`, `model_type`, `version`, `sample_conifg`, `model_config`. 
+* `id` : This is a unique id given to a model based on model_config, sample_conifg, version, and model_type. Users can use it to check if the model matches to another model they have defined.
+* `model_type` : Model type tells us what kind of model it is. This in this case it outputs **Linear Model** 
+* `version` : In case users want to improvise on models, they can keep track of model by its version. As the version changes the unique hash in the `id` also changes.
+* `sample_conifg` : It stores values of the sampler configuration set by user for this particular model.
+* `model_config` : It stores values of the model configuration set by user for this particular model.
+
+```{code-cell} ipython3
+idata = model.fit()
+```
+
+After fitting the model, we can probably save it to share the model as a file so one can use it again.
+To `save` or `load`, we can quickly call methods for respective tasks with the following syntax.
+
+```{code-cell} ipython3
+:tags: []
+
+fname = "/Users/mac/Desktop/mymodel.nc"
+model.save(fname)
+```
+
+This saves a file at the given path, and the name <br>
+A `.nc` file that stores the inference data of the model.
+
++++
+
+Now we create a new model named `model_2` which is same as `model` but instead of fitting it we load it's idata.
+
+```{code-cell} ipython3
+data, model_config, sampler_config = LinearModel.create_sample_input()
+model_2 = LinearModel(model_config, sampler_config, data)
+# loading
+model_2.load(fname)
+```
+
+The `predict()` method allows users to do a posterior predcit with the fitted model on new data.<br>
+Our first task is to create data on which we need to predict.
+
+```{code-cell} ipython3
+x_pred = np.random.uniform(low=0, high=1, size=100)
+prediction_data = pd.DataFrame({"input": x_pred})
+```
+
+Now we predict with `model_2` and save the mean and samples in `pred_mean` and `pred_samples` respectively
+
+```{code-cell} ipython3
+pred_mean = model_2.predict(prediction_data)
+# samples
+pred_samples = model_2.predict_posterior(prediction_data)
+```
+
+After using the `predict()`, we can plot our data and see graphically how satisfactory our `LinearModel` is
+
++++
+
+For which we plot our results
+
+```{code-cell} ipython3
+plt.figure(figsize=(7, 7))
+plt.plot(data["input"], data["output"], "x", label="data")
+plt.plot(prediction_data.input, pred_mean["y_model"], "bo", label="predict", color="r")
+plt.title("Posterior predictive regression lines")
+plt.legend(loc=0)
+plt.xlabel("x")
+plt.ylabel("y");
+```
+
```

Comment:
You need to add youself as author as well as adding the watermark title and page footer. See https://docs.pymc.io/en/latest/contributing/jupyter_style.html#authorship-and-attribution and until the end (skipping the references section).

Reply:
The [watermark header](https://docs.pymc.io/en/latest/contributing/jupyter_style.html#watermark) and [page footer](https://docs.pymc.io/en/latest/contributing/jupyter_style.html#epilogue) are still missing.